### PR TITLE
feat(location): add map component

### PR DESCRIPTION
**Pull request by eugenederii**

### CODE REVIEW

**Review by VadimTarasenko — CHANGES_REQUESTED** [review]

File: desktop.ini [hunk 1]
```diff
@@ -0,0 +1,2 @@
+[.ShellClassInfo]
+IconResource=C:\Windows\System32\SHELL32.dll,77
```
Comment (VadimTarasenko) [inline desktop.ini:2]: remove this file

File: src/modules/core/components/map/index.ts
Comment (VadimTarasenko) [inline src/modules/core/components/map/index.ts:1]: add `export { Map } from './map;`

File: src/modules/core/components/map/map.module.scss [hunk 2]
```diff
@@ -0,0 +1,99 @@
+@use '@/styles/variables.module';
+
+.map_container {
+  position: relative;
+  height: 500px;
+  display: flex;
+}
+
+// marker with photo
+.custom_icon_container {
```
Comment (VadimTarasenko) [inline src/modules/core/components/map/map.module.scss:10]: name all css classes in camelCase style - `customIconContainer`. And mayble remove `custom_` prefix in the class name. So, just `iconContainer`, `avatar` and etc...

File: src/modules/core/components/map/map.tsx [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+import avatar from './avatar.png';
+
+const customMarkerIcon = divIcon({
+  html: `<div class=${styles.custom_icon_container}>
+            <img class=${styles.custom_avatar} src=${avatar.src} alt='Marker' />
+          </div>`,
+  iconSize: [0, 0],
+  iconAnchor: [20, 50],
+  className: '',
+});
```
Comment (VadimTarasenko) [inline src/modules/core/components/map/map.tsx:24]: rename to `proUserMarkerIcon` and move to separate file - `~/Project/src/modules/location/constants/map.constants.ts`

File: src/modules/core/components/map/map.tsx [hunk 4]
```diff
@@ -0,0 +1,90 @@
...
+
+const customMarker = divIcon({
+  html: `<div class=${styles.custom_marker_container}>
+            <div class=${styles.custom_marker_animation}></div>
+            <div class=${styles.custom_marker_center}></div>
+          </div>`,
+});
```
Comment (VadimTarasenko) [inline src/modules/core/components/map/map.tsx:31]: rename to `viewerUserMarkerIcon` and move to `~/Project/src/modules/location/constants/map.constants.ts`

File: src/modules/core/components/map/map.tsx [hunk 5]
```diff
@@ -0,0 +1,90 @@
...
+
+export const Map: FC = () => {
+  const [position, setPosition] = useState<null | [number, number]>(null);
+
+  const mapRef = useRef<L.Map>(null);
+
+  useEffect(() => {
+    navigator.permissions.query({ name: 'geolocation' }).then((result) => {
```
Comment (VadimTarasenko) [inline src/modules/core/components/map/map.tsx:39]: Probably we don't need this line. Please, check if it works without 39 line.

File: src/modules/core/components/map/map.tsx [hunk 6]
```diff
@@ -0,0 +1,90 @@
...
+    mapRef.current?.zoomIn();
+  };
+
+  const handleZoomOut = () => {
+    mapRef.current?.zoomOut();
+  };
+
+  const URL = 'eugenederii/clkls8pny000c01qx0bnr2mhl';
+  const AccessToken =
+    'pk.eyJ1IjoiZXVnZW5lZGVyaWkiLCJhIjoiY2xrbHMxY2t5MDZucjNlb3h0Mmc4OGVlcSJ9.stj0ywdapaAI46qlOoFV9A';
```
Comment (VadimTarasenko) [inline src/modules/core/components/map/map.tsx:56]: remove URL and AccessToken

File: src/modules/core/components/map/map.tsx [hunk 7]
```diff
@@ -0,0 +1,90 @@
...
+        setPosition([pos.coords.latitude, pos.coords.longitude]);
+      });
+    });
+  }, []);
+
+  const handleZoomIn = () => {
+    mapRef.current?.zoomIn();
+  };
+
+  const handleZoomOut = () => {
```
Comment (VadimTarasenko) [inline src/modules/core/components/map/map.tsx:50]: wrap into `useCallback`:

```ts
const handleZoomOut = useCallback(() => {
  mapRef.current?.zoomOut();
}, [])
```

And handleZoomIn too.

File: src/modules/core/components/map/map.tsx [hunk 8]
```diff
@@ -0,0 +1,90 @@
...
+      />
+      <MapContainer
+        center={[50.517483, 30.495037]}
+        zoom={13}
+        scrollWheelZoom={true}
+        zoomControl={false}
+        ref={mapRef}
+      >
+        <TileLayer
+          url={`https://api.mapbox.com/styles/v1/${URL}/tiles/{z}/{x}/{y}?access_token=${AccessToken}`}
```
Comment (VadimTarasenko) [inline src/modules/core/components/map/map.tsx:73]: do we still need mapbox here?

File: src/modules/core/components/map/map.tsx [hunk 9]
```diff
@@ -0,0 +1,90 @@
...
+        />
+        {position && (
+          <>
+            <Marker icon={customMarker} position={position}>
+              <Popup>
+                A pretty CSS3 popup. <br /> Easily customizable.
+              </Popup>
+            </Marker>
+            <Marker icon={customMarkerIcon} position={[50.517483, 30.495037]}>
+              <Popup>ddd</Popup>
```
Comment (VadimTarasenko) [inline src/modules/core/components/map/map.tsx:83]: remove popups as we don't need them at the moment

File: src/modules/core/components/map/map.tsx [hunk 10]
```diff
@@ -0,0 +1,90 @@
...
+            </Marker>
+          </>
+        )}
+      </MapContainer>
+    </div>
+  );
+};
```
Comment (VadimTarasenko) [inline src/modules/core/components/map/map.tsx:90]: This component should be in `location` module. So, you have to create folder for location module and put this component there:

`~/Project/src/modules/location/components/map/*`

File: src/modules/core/components/zoom/index.ts
Comment (VadimTarasenko) [inline src/modules/core/components/zoom/index.ts:1]: add `export { Zoom } from './zoom';`

File: src/modules/core/components/zoom/zoom.module.scss [hunk 11]
```diff
@@ -0,0 +1,41 @@
+.zoom_control {
+  width: 32px;
+  height: 65px;
+  display: flex;
+  flex-direction: column;
+  box-shadow: 0px 4px 15px 0px rgba(37, 74, 165, 0.1);
```
Comment (VadimTarasenko) [inline src/modules/core/components/zoom/zoom.module.scss:6]: use scss variables from file 'src/styles/variables.module.scss'. In this case use `variables.$main-box-shadow` and check other places where you can use other variables.

File: src/modules/core/components/zoom/zoom.tsx [hunk 12]
```diff
@@ -0,0 +1,24 @@
+import { FC } from 'react';
+import styles from './zoom.module.scss';
+import { Icon } from '@/modules/core/components';
+import { ZoomProps } from './zoom.interface';
+import clsx from 'clsx';
+
+export const Zoom: FC<ZoomProps> = ({
```
Comment (VadimTarasenko) [inline src/modules/core/components/zoom/zoom.tsx:7]: Rename this component to `ZoomControl` and move to location module

**Review by VadimTarasenko — CHANGES_REQUESTED** [review]

File: src/modules/core/components/emoji/emoji.module.scss
Comment (VadimTarasenko) [inline src/modules/core/components/emoji/emoji.module.scss:1]: remove this file

File: src/modules/core/components/zoom/zoom.module.scss [hunk 13]
```diff
@@ -0,0 +1,43 @@
+@use '@/styles/variables.module';
+
+.zoom_control {
```
Comment (VadimTarasenko) [inline src/modules/core/components/zoom/zoom.module.scss:3]: use camelCase for classes naming

File: src/modules/core/components/zoom/zoom.stories.tsx [hunk 14]
```diff
@@ -0,0 +1,14 @@
+import type { Meta, StoryObj } from '@storybook/react';
+
+import { ZoomControl } from './zoom';
+
+const meta: Meta<typeof ZoomControl> = {
+  title: 'Core UI/Zoom',
```
Comment (VadimTarasenko) [inline src/modules/core/components/zoom/zoom.stories.tsx:6]: change to `Location UI/ZoomControl` and move `zoom` folder to the `modules/location/components` and rename folder name to `zoom-control` and files like `zoom-control.tsx`...

File: src/modules/location/map/index.ts [hunk 15]
```diff
@@ -0,0 +1 @@
+export { Map } from './map';
```
Comment (VadimTarasenko) [inline src/modules/location/map/index.ts:1]: you have to create `components` folder in the `location` and then move map to the components

**Review by VadimTarasenko — CHANGES_REQUESTED** [review]

File: src/modules/location/map/map.stories.tsx [hunk 16]
```diff
@@ -0,0 +1,15 @@
+import type { Meta, StoryObj } from '@storybook/react';
+import 'leaflet/dist/leaflet.css';
+
+import { Map } from './map';
+
+const meta: Meta<typeof Map> = {
+  title: 'Core UI/Map',
```
Comment (VadimTarasenko) [inline src/modules/location/map/map.stories.tsx:7]: rename title to `Location UI/Map`

**Review by VadimTarasenko — CHANGES_REQUESTED** [review]

File: src/modules/location/constants/map.constants.ts [hunk 17]
```diff
@@ -0,0 +1,22 @@
+import { divIcon } from 'leaflet';
+import styles from '@/modules/location/components/map/map.module.scss';
+
+const ICON_ANCHOR_X = 20;
+const ICON_ANCHOR_Y = 50;
+
+export const proUserMarkerIcon = (urlAvatar: string) =>
+  divIcon({
+    html: `<div class=${styles.iconContainer}>
```
[the remark this answers is not in the document]
Comment (eugenederii) [reply src/modules/location/constants/map.constants.ts:9]: No, I havnt found another way to use this

**Review by VadimTarasenko — APPROVED** [review]

**Pull request merged**